File: typescript-connector/src/models/Place.ts
// Class from DFC Business Ontology: #Place
import { SemanticObject } from "../core/SemanticObject.js";
import { WhereSubject, type WhereSubjectParams } from "./WhereSubject.js";
import type { SaleSession } from "./SaleSession.js";

export interface PlaceParams extends WhereSubjectParams {
  hosts?: (SaleSession | string)[];
}
export class Place extends WhereSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Place";
  }

  hosts?: (SaleSession | string)[];

  constructor(
    semanticId: string,
    params?: PlaceParams,
  ) {
    super(semanticId, params);
    this.hosts = params?.hosts;
    this.semanticType = Place.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:hosts", () => this.hosts);
  }
  static {
    SemanticObject.typeRegistry.set(Place.SEMANTIC_TYPE, Place);
  }
}

File: ruby-gem/lib/models/place.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Place
require_relative '../core/semantic_object'
require_relative 'where_subject'


module DfcLinkmlConnector
  module Models
    class Place < WhereSubject
      SEMANTIC_TYPE = "dfc-b:Place".freeze

      # @return [String]
      attr_accessor :hosts

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, hosts: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, hosts: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @hosts = hosts
        self.semanticType = "dfc-b:Place"
        registerSemanticProperty("dfc-b:Place:hosts", &method("hosts")).valueSetter = method("hosts=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
